librz/bin/format/luac/luac_bin.c

```diff
@@ -0,0 +1,306 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// SPDX-FileCopyrightText: 2021 Heersin <teablearcher@gmail.com>
+
+#include "luac_common.h"
+
+void luac_add_section(RzList *section_list, char *name, ut64 offset, ut32 size, bool is_func) {
+	RzBinSection *bin_sec = RZ_NEW0(RzBinSection);
+	if (!bin_sec) {
+		return;
+	}
+
+	bin_sec->name = rz_str_new(name);
+	bin_sec->vaddr = bin_sec->paddr = offset;
+	bin_sec->size = bin_sec->vsize = size;
+	bin_sec->add = true;
+	bin_sec->is_data = false;
+	bin_sec->bits = is_func ? sizeof(LUA_INSTRUCTION) * 8 : 8;
+	// bin_sec->has_strings = !is_func;
+	bin_sec->has_strings = false;
+	bin_sec->arch = rz_str_new("luac");
+
+	if (is_func) {
+		bin_sec->perm = RZ_PERM_R | RZ_PERM_X;
+	} else {
+		bin_sec->perm = RZ_PERM_R;
+	}
+
+	rz_list_append(section_list, bin_sec);
+}
+
+void luac_add_symbol(RzList *symbol_list, char *name, ut64 offset, ut64 size, const char *type) {
+	RzBinSymbol *bin_sym = RZ_NEW0(RzBinSymbol);
+	if (!bin_sym) {
+		return;
+	}
+
+	bin_sym->name = rz_str_new(name);
+	bin_sym->vaddr = bin_sym->paddr = offset;
+	bin_sym->size = size;
+	bin_sym->type = type;
+
+	rz_list_append(symbol_list, bin_sym);
+}
+
+void luac_add_entry(RzList *entry_list, ut64 offset, int entry_type) {
+	RzBinAddr *entry = RZ_NEW0(RzBinAddr);
+	if (!entry) {
+		return;
+	}
+
+	entry->vaddr = offset;
```

Comment:
is there no difference between virtual address and file offset in luac files? Just asking as I don't know the LUAC format.

Reply:
Yes, I don't really think there is a concept of virtual address of it. Its virtual machine is a loop to fetch and execute the opcode, likes ESIL VM in rizin. Should the virtual address output be disabled ?